sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -345,18 +346,25 @@ static void _on_sock_udp_evt(sock_udp_t *sock, sock_async_flags_t type, void *ar
             cursor += res;
         }
 
-        /* make sure we reply with the same address that the request was destined for */
+        /* make sure we reply with the same address that the request was
+         * destined for -- except in the multicast case */
+        sock_udp_aux_tx_t *aux_out_ptr;
         sock_udp_aux_tx_t aux_out = {
             .flags = SOCK_AUX_SET_LOCAL,
             .local = aux_in.local,
         };
+        if (_ep_is_multicast(&aux_in.local)) {
+            aux_out_ptr = NULL;
```

Comment:
Probably I'm too late, but a comment about the semantics of setting this pointer to NULL here would have been nice.

Reply:
Good point, given it takes several indirections before this wins up in a function where one can read the documentation.

Comment added, let's see how mad bors will get at me for throwing logs before its merge train.